lib/pprof/pprof.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pprof
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	_ "net/http/pprof"
+	"net/url"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/vmware/vic/lib/metadata"
+	"github.com/vmware/vic/pkg/vsphere/extraconfig"
+)
+
+type PprofPort int
+
+const basePort = 6060
+
+const (
+	VCHInitPort PprofPort = iota
+	ImagecPort
+	VicadminPort
+	DockerPort
+	PortlayerPort
+	maxPort
+)
+
+var (
+	vchConfig metadata.VirtualContainerHostConfigSpec
+)
+
+func init() {
+	// load the vch config
+	// TODO: Optimize this to just pull the fields we need...
+	src, err := extraconfig.GuestInfoSource()
+	if err != nil {
+		log.Errorf("Unable to load configuration from guestinfo")
+		return
+	}
+	extraconfig.Decode(src, &vchConfig)
+}
+
+func StartPprof(name string, component PprofPort) error {
+	port := component + basePort
+	if component != ImagecPort {
```

Comment:
I we're dropping imagec because it was causing issues, then let's just remove it from the set of components for which we're launching pprof. Filtering it here makes little sense.